collaboration-engine/src/main/java/com/vaadin/collaborationengine/CollaborationMessageList.java

```diff
@@ -291,11 +296,63 @@ public void setImageProvider(ImageProvider imageProvider) {
      *
      * @return the current image provider callback, or <code>null</code> if no
      *         callback is set
+     * @deprecated Use {@link #setImageHandler(ImageHandler)} instead.
      */
+    @Deprecated(since = "6.5", forRemoval = true)
     public ImageProvider getImageProvider() {
         return imageProvider;
     }
 
+    /**
+     * Sets an image handler callback for dynamically loading avatar images for
+     * a given user. The image can be loaded on-demand from a database or using
+     * any other source of IO streams.
+     * <p>
+     * If no image handler is defined, then the image URL defined by
+     * {@link UserInfo#getImage()} is directly passed to the browser. This means
+     * that avatar images need to be available as static files or served
+     * dynamically from a custom servlet. This is the default.
+     *
+     * Usage example:
+     *
+     * <pre>
+     * collaborationMessageList.setImageHandler(userInfo -> {
+     *     DownloadHandler downloadHandler = DownloadHandler
+     *             .fromInputStream(context -> {
+     *                 context.setFileName("avatar_" + userInfo.getId());
+     *                 context.setContentType("image/png");
+     *                 User userEntity = userRepository
+     *                         .findById(userInfo.getId());
+     *                 byte[] profilePicture = userEntity.getProfilePicture();
+     *                 return new ByteArrayInputStream(profilePicture);
```

Comment:
When I tried this example it complained that the `ByteArrayInputStream` cannot be converted to `DownloadResponse`. Do I missed something?